Author: Adam Majer <amajer@suse.de>
Date: Tue Jul  4 15:54:34 CEST 2017
Summary: Generate versioned binaries

Generate versioned binaries and install paths
so we can allow concurrent installations and
management via update_alternatives.

This is also important for generation of binary
modules for multiple versions of NodeJS
Index: node-v9.1.0/Makefile
===================================================================
--- node-v9.1.0.orig/Makefile
+++ node-v9.1.0/Makefile
@@ -45,10 +45,10 @@ BUILDTYPE_LOWER := $(shell echo $(BUILDT
 EXEEXT := $(shell $(PYTHON) -c \
 		"import sys; print('.exe' if sys.platform == 'win32' else '')")
 
-NODE_EXE = node$(EXEEXT)
+NODE_EXE = node9$(EXEEXT)
 NODE ?= ./$(NODE_EXE)
 NODE_G_EXE = node_g$(EXEEXT)
-NPM ?= ./deps/npm/bin/npm-cli.js
+NPM ?= ./deps/npm9/bin/npm-cli.js
 
 # Flags for packaging.
 BUILD_DOWNLOAD_FLAGS ?= --download=all
Index: node-v9.1.0/tools/install.py
===================================================================
--- node-v9.1.0.orig/tools/install.py
+++ node-v9.1.0/tools/install.py
@@ -78,7 +78,7 @@ def install(paths, dst): map(lambda path
 def uninstall(paths, dst): map(lambda path: try_remove(path, dst), paths)
 
 def npm_files(action):
-  target_path = sysconfig.get_config_var("LIB") + '/node_modules/npm/'
+  target_path = sysconfig.get_config_var("LIB") + '/node_modules/npm9/'
 
   # don't install npm if the target path is a symlink, it probably means
   # that a dev version of npm is installed there
@@ -92,20 +92,20 @@ def npm_files(action):
     action(paths, target_path + dirname[9:] + '/')
 
   # create/remove symlink
-  link_path = abspath(install_path, 'bin/npm')
+  link_path = abspath(install_path, 'bin/npm9')
   if action == uninstall:
-    action([link_path], 'bin/npm')
+    action([link_path], 'bin/npm9')
   elif action == install:
-    try_symlink("../" + sysconfig.get_config_var("LIB") + '/node_modules/npm/bin/npm-cli.js',link_path)
+    try_symlink("../" + sysconfig.get_config_var("LIB") + '/node_modules/npm9/bin/npm-cli.js',link_path)
   else:
     assert(0) # unhandled action type
 
   # create/remove symlink
-  link_path = abspath(install_path, 'bin/npx')
+  link_path = abspath(install_path, 'bin/npx9')
   if action == uninstall:
-    action([link_path], 'bin/npx')
+    action([link_path], 'bin/npx9')
   elif action == install:
-    try_symlink('../lib/node_modules/npm/bin/npx-cli.js', link_path)
+    try_symlink("../" + sysconfig.get_config_var("LIB") + '/node_modules/npm9/bin/npx-cli.js', link_path)
   else:
     assert(0) # unhandled action type
 
@@ -119,7 +119,7 @@ def subdir_files(path, dest, action):
 
 def files(action):
   is_windows = sys.platform == 'win32'
-  output_file = 'node'
+  output_file = 'node9'
   output_prefix = 'out/Release/'
 
   if 'false' == variables.get('node_shared'):
@@ -144,7 +144,7 @@ def files(action):
     action(['out/Release/node.d'], sysconfig.get_config_var("LIB") + '/dtrace/node.d')
 
   # behave similarly for systemtap
-  action(['src/node.stp'], 'share/systemtap/tapset/')
+  action(['src/node.stp'], 'share/systemtap/tapset/node9.stp')
 
   action(['deps/v8/tools/gdbinit'], 'share/doc/node/')
   action(['deps/v8/tools/lldbinit'], 'share/doc/node/')
@@ -153,7 +153,7 @@ def files(action):
   if 'freebsd' in sys.platform or 'openbsd' in sys.platform:
     action(['doc/node.1'], 'man/man1/')
   else:
-    action(['doc/node.1'], 'share/man/man1/')
+    action(['doc/node.1'], 'share/man/man1/node9.1')
 
   if 'true' == variables.get('node_install_npm'): npm_files(action)
 
@@ -169,28 +169,28 @@ def headers(action):
     'src/node_buffer.h',
     'src/node_object_wrap.h',
     'src/node_version.h',
-  ], 'include/node/')
+  ], 'include/node9/')
 
   # Add the expfile that is created on AIX
   if sys.platform.startswith('aix'):
-    action(['out/Release/node.exp'], 'include/node/')
+    action(['out/Release/node.exp'], 'include/node9/')
 
-  subdir_files('deps/v8/include', 'include/node/', action)
+  subdir_files('deps/v8/include', 'include/node9/', action)
 
   if 'false' == variables.get('node_shared_libuv'):
-    subdir_files('deps/uv/include', 'include/node/', action)
+    subdir_files('deps/uv/include', 'include/node9/', action)
 
   if 'true' == variables.get('node_use_openssl') and \
      'false' == variables.get('node_shared_openssl'):
-    subdir_files('deps/openssl/openssl/include/openssl', 'include/node/openssl/', action)
-    subdir_files('deps/openssl/config/archs', 'include/node/openssl/archs', action)
-    action(['deps/openssl/config/opensslconf.h'], 'include/node/openssl/')
+    subdir_files('deps/openssl/openssl/include/openssl', 'include/node9/openssl/', action)
+    subdir_files('deps/openssl/config/archs', 'include/node9/openssl/archs', action)
+    action(['deps/openssl/config/opensslconf.h'], 'include/node9/openssl/')
 
   if 'false' == variables.get('node_shared_zlib'):
     action([
       'deps/zlib/zconf.h',
       'deps/zlib/zlib.h',
-    ], 'include/node/')
+    ], 'include/node9/')
 
 def run(args):
   global node_prefix, install_path, target_defaults, variables
Index: node-v9.1.0/doc/node.1
===================================================================
--- node-v9.1.0.orig/doc/node.1
+++ node-v9.1.0/doc/node.1
@@ -26,12 +26,12 @@
 
 .SH NAME
 
-node \- Server-side JavaScript runtime
+node9 \- Server-side JavaScript runtime
 
 
 .SH SYNOPSIS
 
-.B node
+.B node9
 .RI [ options ]
 .RI [ v8\ options ]
 .RI [ script.js \ |
@@ -43,14 +43,14 @@ node \- Server-side JavaScript runtime
 .B [--]
 .RI [ arguments ]
 .br
-.B node debug
+.B node9 debug
 .RI [ script.js " | "
 .B \-e
 .RI \&" script \&"\ |
 .IR <host>:<port> ]
 .I ...
 .br
-.B node
+.B node9
 .RB [ \-\-v8-options ]
 
 Execute without arguments to start the REPL.
Index: node-v9.1.0/src/node.stp
===================================================================
--- node-v9.1.0.orig/src/node.stp
+++ node-v9.1.0/src/node.stp
@@ -19,7 +19,7 @@
 // OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE
 // USE OR OTHER DEALINGS IN THE SOFTWARE.
 
-probe node_net_server_connection = process("node").mark("net__server__connection")
+probe node_net_server_connection = process("node9").mark("net__server__connection")
 {
   remote = user_string($arg2);
   port = $arg3;
@@ -32,7 +32,7 @@ probe node_net_server_connection = proce
     fd);
 }
 
-probe node_net_stream_end = process("node").mark("net__stream__end")
+probe node_net_stream_end = process("node9").mark("net__stream__end")
 {
   remote = user_string($arg2);
   port = $arg3;
@@ -45,7 +45,7 @@ probe node_net_stream_end = process("nod
     fd);
 }
 
-probe node_http_server_request = process("node").mark("http__server__request")
+probe node_http_server_request = process("node9").mark("http__server__request")
 {
   remote = user_string($arg3);
   port = $arg4;
@@ -62,7 +62,7 @@ probe node_http_server_request = process
     fd);
 }
 
-probe node_http_server_response = process("node").mark("http__server__response")
+probe node_http_server_response = process("node9").mark("http__server__response")
 {
   remote = user_string($arg2);
   port = $arg3;
@@ -75,7 +75,7 @@ probe node_http_server_response = proces
     fd);
 }
 
-probe node_http_client_request = process("node").mark("http__client__request")
+probe node_http_client_request = process("node9").mark("http__client__request")
 {
   remote = user_string($arg3);
   port = $arg4;
@@ -92,7 +92,7 @@ probe node_http_client_request = process
     fd);
 }
 
-probe node_http_client_response = process("node").mark("http__client__response")
+probe node_http_client_response = process("node9").mark("http__client__response")
 {
   remote = user_string($arg2);
   port = $arg3;
@@ -105,7 +105,7 @@ probe node_http_client_response = proces
     fd);
 }
 
-probe node_gc_start = process("node").mark("gc__start")
+probe node_gc_start = process("node9").mark("gc__start")
 {
   scavenge = 1 << 0;
   compact = 1 << 1;
@@ -125,7 +125,7 @@ probe node_gc_start = process("node").ma
     flags);
 }
 
-probe node_gc_stop = process("node").mark("gc__stop")
+probe node_gc_stop = process("node9").mark("gc__stop")
 {
   scavenge = 1 << 0;
   compact = 1 << 1;
Index: node-v9.1.0/deps/npm/man/man1/npm.1
===================================================================
--- node-v9.1.0.orig/deps/npm/man/man1/npm.1
+++ node-v9.1.0/deps/npm/man/man1/npm.1
@@ -1,11 +1,11 @@
 .TH "NPM" "1" "October 2017" "" ""
 .SH "NAME"
-\fBnpm\fR \- javascript package manager
+\fBnpm9\fR \- javascript package manager
 .SH SYNOPSIS
 .P
 .RS 2
 .nf
-npm <command> [args]
+npm9 <command> [args]
 .fi
 .RE
 .SH VERSION
@@ -21,16 +21,16 @@ It is extremely configurable to support
 Most commonly, it is used to publish, discover, install, and develop node
 programs\.
 .P
-Run \fBnpm help\fP to get a list of available commands\.
+Run \fBnpm9 help\fP to get a list of available commands\.
 .SH INTRODUCTION
 .P
 You probably got npm because you want to install stuff\.
 .P
-Use \fBnpm install blerg\fP to install the latest version of "blerg"\.  Check out
-npm help \fBnpm\-install\fP for more info\.  It can do a lot of stuff\.
+Use \fBnpm9 install blerg\fP to install the latest version of "blerg"\.  Check out
+npm9 help \fBnpm\-install\fP for more info\.  It can do a lot of stuff\.
 .P
-Use the \fBnpm search\fP command to show everything that's available\.
-Use \fBnpm ls\fP to show everything you've installed\.
+Use the \fBnpm9 search\fP command to show everything that's available\.
+Use \fBnpm9 ls\fP to show everything you've installed\.
 .SH DEPENDENCIES
 .P
 If a package references to another package with a git URL, npm depends
@@ -48,7 +48,7 @@ the node\-gyp repository \fIhttps://gith
 the node\-gyp Wiki \fIhttps://github\.com/TooTallNate/node\-gyp/wiki\fR\|\.
 .SH DIRECTORIES
 .P
-See npm help 5 \fBnpm\-folders\fP to learn about where npm puts stuff\.
+See npm9 help 5 \fBnpm\-folders\fP to learn about where npm puts stuff\.
 .P
 In particular, npm has two modes of operation:
 .RS 0
@@ -80,19 +80,19 @@ Make a package\.json file\.  See npm hel
 link:
 For linking your current working code into Node's path, so that you
 don't have to reinstall every time you make a change\.  Use
-\fBnpm link\fP to do this\.
+\fBnpm9 link\fP to do this\.
 .IP \(bu 2
 install:
 It's a good idea to install things if you don't need the symbolic link\.
 Especially, installing other peoples code from the registry is done via
-\fBnpm install\fP
+\fBnpm9 install\fP
 .IP \(bu 2
 adduser:
 Create an account or log in\.  Credentials are stored in the
 user config file\.
 .IP \(bu 2
 publish:
-Use the \fBnpm publish\fP command to upload your code to the registry\.
+Use the \fBnpm9 publish\fP command to upload your code to the registry\.
 
 .RE
 .SH CONFIGURATION
@@ -133,14 +133,14 @@ lib/utils/config\-defs\.js\.  These must
 
 .RE
 .P
-See npm help 7 \fBnpm\-config\fP for much much more information\.
+See npm9 help 7 \fBnpm\-config\fP for much much more information\.
 .SH CONTRIBUTIONS
 .P
 Patches welcome!
 .RS 0
 .IP \(bu 2
 code:
-Read through npm help 7 \fBnpm\-coding\-style\fP if you plan to submit code\.
+Read through npm9 help 7 \fBnpm\-coding\-style\fP if you plan to submit code\.
 You don't have to agree with it, but you do have to follow it\.
 .IP \(bu 2
 docs:
@@ -150,7 +150,7 @@ file in the "doc" folder\.  (Don't worry
 .RE
 .P
 Contributors are listed in npm's \fBpackage\.json\fP file\.  You can view them
-easily by doing \fBnpm view npm contributors\fP\|\.
+easily by doing \fBnpm9 view npm contributors\fP\|\.
 .P
 If you would like to contribute, but don't know what to work on, read
 the contributing guidelines and check the issues list\.
@@ -185,21 +185,21 @@ i@izs\.me
 .SH SEE ALSO
 .RS 0
 .IP \(bu 2
-npm help help
+npm9 help help
 .IP \(bu 2
 README
 .IP \(bu 2
-npm help 5 package\.json
+npm9 help 5 package\.json
 .IP \(bu 2
-npm help install
+npm9 help install
 .IP \(bu 2
-npm help config
+npm9 help config
 .IP \(bu 2
-npm help 7 config
+npm9 help 7 config
 .IP \(bu 2
-npm help 5 npmrc
+npm9 help 5 npmrc
 .IP \(bu 2
-npm help 7 index
+npm9 help 7 index
 
 .RE
 
Index: node-v9.1.0/node.gyp
===================================================================
--- node-v9.1.0.orig/node.gyp
+++ node-v9.1.0/node.gyp
@@ -20,7 +20,7 @@
     'node_shared_openssl%': 'false',
     'node_v8_options%': '',
     'node_enable_v8_vtunejit%': 'false',
-    'node_core_target_name%': 'node',
+    'node_core_target_name%': 'node9',
     'library_files': [
       'lib/internal/bootstrap_node.js',
       'lib/async_hooks.js',
@@ -498,10 +498,10 @@
             {
               'action_name': 'node_dtrace_provider_o',
               'inputs': [
-                '<(OBJ_DIR)/node/src/node_dtrace.o',
+                '<(OBJ_DIR)/node9/src/node_dtrace.o',
               ],
               'outputs': [
-                '<(OBJ_DIR)/node/src/node_dtrace_provider.o'
+                '<(OBJ_DIR)/node9/src/node_dtrace_provider.o'
               ],
               'action': [ 'dtrace', '-G', '-xnolibs', '-s', 'src/node_provider.d',
                 '<@(_inputs)', '-o', '<@(_outputs)' ]
@@ -551,7 +551,7 @@
                 '<(SHARED_INTERMEDIATE_DIR)/v8constants.h'
               ],
               'outputs': [
-                '<(OBJ_DIR)/node/src/node_dtrace_ustack.o'
+                '<(OBJ_DIR)/node9/src/node_dtrace_ustack.o'
               ],
               'conditions': [
                 [ 'target_arch=="ia32" or target_arch=="arm"', {
@@ -612,9 +612,9 @@
       ],
 
       'variables': {
-        'OBJ_PATH': '<(OBJ_DIR)/node/src',
-        'OBJ_GEN_PATH': '<(OBJ_DIR)/node/gen',
-        'OBJ_TRACING_PATH': '<(OBJ_DIR)/node/src/tracing',
+        'OBJ_PATH': '<(OBJ_DIR)/node9/src',
+        'OBJ_GEN_PATH': '<(OBJ_DIR)/node9/gen',
+        'OBJ_TRACING_PATH': '<(OBJ_DIR)/node9/src/tracing',
         'OBJ_SUFFIX': 'o',
         'OBJ_SEPARATOR': '/',
         'conditions': [
@@ -750,7 +750,7 @@
     ['OS=="aix"', {
       'targets': [
         {
-          'target_name': 'node',
+          'target_name': 'node9',
           'conditions': [
             ['node_shared=="true"', {
               'type': 'shared_library',
Index: node-v9.1.0/deps/npm/bin/npm-cli.js
===================================================================
--- node-v9.1.0.orig/deps/npm/bin/npm-cli.js
+++ node-v9.1.0/deps/npm/bin/npm-cli.js
@@ -1,4 +1,4 @@
-#!/usr/bin/env node
+#!/usr/bin/env node9
 ;(function () { // wrapper in case we're in module_context mode
   // windows: running "npm blah" in this folder will invoke WSH, not node.
   /*global WScript*/
Index: node-v9.1.0/src/node_main.cc
===================================================================
--- node-v9.1.0.orig/src/node_main.cc
+++ node-v9.1.0/src/node_main.cc
@@ -101,6 +101,7 @@ int main(int argc, char *argv[]) {
 #endif
   // Disable stdio buffering, it interacts poorly with printf()
   // calls elsewhere in the program (e.g., any logging from V8.)
+  setenv("NODE_VERSION", "9", 0);
   setvbuf(stdout, nullptr, _IONBF, 0);
   setvbuf(stderr, nullptr, _IONBF, 0);
   return node::Start(argc, argv);
Index: node-v9.1.0/deps/npm/man/man1/npx.1
===================================================================
--- node-v9.1.0.orig/deps/npm/man/man1/npx.1
+++ node-v9.1.0/deps/npm/man/man1/npx.1
@@ -1,32 +1,32 @@
 .TH "NPX" "1" "August 2017" "npx@9.5.0" "User Commands"
 .SH "NAME"
-\fBnpx\fR \- execute npm package binaries
+\fBnpx9\fR \- execute npm package binaries
 .SH SYNOPSIS
 .P
-\fBnpx [options] <command>[@version] [command\-arg]\.\.\.\fP
+\fBnpx9 [options] <command>[@version] [command\-arg]\.\.\.\fP
 .P
-\fBnpx [options] [\-p|\-\-package <pkg>]\.\.\. <command> [command\-arg]\.\.\.\fP
+\fBnpx9 [options] [\-p|\-\-package <pkg>]\.\.\. <command> [command\-arg]\.\.\.\fP
 .P
-\fBnpx [options] \-c '<command\-string>'\fP
+\fBnpx9 [options] \-c '<command\-string>'\fP
 .P
-\fBnpx \-\-shell\-auto\-fallback [shell]\fP
+\fBnpx9 \-\-shell\-auto\-fallback [shell]\fP
 .SH INSTALL
 .P
-\fBnpm install \-g npx\fP
+\fBnpm9 install \-g npx\fP
 .SH DESCRIPTION
 .P
 Executes \fB<command>\fP either from a local \fBnode_modules/\.bin\fP, or from a central cache, installing any packages needed in order for \fB<command>\fP to run\.
 .P
-By default, \fBnpx\fP will check whether \fB<command>\fP exists in \fB$PATH\fP, or in the local project binaries, and execute that\. If \fB<command>\fP is not found, it will be installed prior to execution\.
+By default, \fBnpx9\fP will check whether \fB<command>\fP exists in \fB$PATH\fP, or in the local project binaries, and execute that\. If \fB<command>\fP is not found, it will be installed prior to execution\.
 .P
-Unless a \fB\-\-package\fP option is specified, \fBnpx\fP will try to guess the name of the binary to invoke depending on the specifier provided\. All package specifiers understood by \fBnpm\fP may be used with \fBnpx\fP, including git specifiers, remote tarballs, local directories, or scoped packages\.
+Unless a \fB\-\-package\fP option is specified, \fBnpx9\fP will try to guess the name of the binary to invoke depending on the specifier provided\. All package specifiers understood by \fBnpm9\fP may be used with \fBnpx9\fP, including git specifiers, remote tarballs, local directories, or scoped packages\.
 .P
 If a full specifier is included, or if \fB\-\-package\fP is used, npx will always use a freshly\-installed, temporary version of the package\. This can also be forced with the \fB\-\-ignore\-existing\fP flag\.
 .RS 0
 .IP \(bu 2
 \fB\-p, \-\-package <package>\fP \- define the package to be installed\. This defaults to the value of \fB<command>\fP\|\. This is only needed for packages with multiple binaries if you want to call one of the other executables, or where the binary name does not match the package name\. If this option is provided \fB<command>\fP will be executed as\-is, without interpreting \fB@version\fP if it's there\. Multiple \fB\-\-package\fP options may be provided, and all the packages specified will be installed\.
 .IP \(bu 2
-\fB\-\-no\-install\fP \- If passed to \fBnpx\fP, it will only try to run \fB<command>\fP if it already exists in the current path or in \fB$prefix/node_modules/\.bin\fP\|\. It won't try to install missing commands\.
+\fB\-\-no\-install\fP \- If passed to \fBnpx9\fP, it will only try to run \fB<command>\fP if it already exists in the current path or in \fB$prefix/node_modules/\.bin\fP\|\. It won't try to install missing commands\.
 .IP \(bu 2
 \fB\-\-cache <path>\fP \- set the location of the npm cache\. Defaults to npm's own cache settings\.
 .IP \(bu 2
@@ -36,7 +36,7 @@ If a full specifier is included, or if \
 .IP \(bu 2
 \fB\-\-shell <string>\fP \- The shell to invoke the command with, if any\.
 .IP \(bu 2
-\fB\-\-shell\-auto\-fallback [<shell>]\fP \- Generates shell code to override your shell's "command not found" handler with one that calls \fBnpx\fP\|\. Tries to figure out your shell, or you can pass its name (either \fBbash\fP, \fBfish\fP, or \fBzsh\fP) as an option\. See below for how to install\.
+\fB\-\-shell\-auto\-fallback [<shell>]\fP \- Generates shell code to override your shell's "command not found" handler with one that calls \fBnpx9\fP\|\. Tries to figure out your shell, or you can pass its name (either \fBbash\fP, \fBfish\fP, or \fBzsh\fP) as an option\. See below for how to install\.
 .IP \(bu 2
 \fB\-\-ignore\-existing\fP \- If this flag is set, npx will not look in \fB$PATH\fP, or in the current package's \fBnode_modules/\.bin\fP for an existing version before deciding whether to install\. Binaries in those paths will still be available for execution, but will be shadowed by any packages requested by this install\.
 .IP \(bu 2
@@ -103,7 +103,7 @@ Debugger listening on ws://127\.0\.0\.1:
 .RE
 .SH SHELL AUTO FALLBACK
 .P
-You can configure \fBnpx\fP to run as your default fallback command when you type something in the command line with an \fB@\fP but the command is not found\. This includes installing packages that were not found in the local prefix either\.
+You can configure \fBnpx9\fP to run as your default fallback command when you type something in the command line with an \fB@\fP but the command is not found\. This includes installing packages that were not found in the local prefix either\.
 .P
 For example:
 .P
@@ -145,7 +145,7 @@ $ source (npx \-\-shell\-auto\-fallback
 .RE
 .SH ACKNOWLEDGEMENTS
 .P
-Huge thanks to Kwyn Meagher \fIhttps://blog\.kwyn\.io\fR for generously donating the package name in the main npm registry\. Previously \fBnpx\fP was used for a Tessel board Neopixels library, which can now be found under \fBnpx\-tessel\fP \fIhttps://npm\.im/npx\-tessel\fR\|\.
+Huge thanks to Kwyn Meagher \fIhttps://blog\.kwyn\.io\fR for generously donating the package name in the main npm registry\. Previously \fBnpx9\fP was used for a Tessel board Neopixels library, which can now be found under \fBnpx\-tessel\fP \fIhttps://npm\.im/npx\-tessel\fR\|\.
 .SH AUTHOR
 .P
 Written by Kat Marchan \fIhttps://github\.com/zkat\fR\|\.
@@ -158,10 +158,10 @@ This work is released by its authors int
 .SH SEE ALSO
 .RS 0
 .IP \(bu 2
-\fBnpm(1)\fP
+\fBnpm9(1)\fP
 .IP \(bu 2
-\fBnpm\-run\-script(1)\fP
+\fBnpm9\-run\-script(1)\fP
 .IP \(bu 2
-\fBnpm\-config(7)\fP
+\fBnpm9\-config(7)\fP
 
 .RE
